### PR TITLE
[FLINK-22152][python] Fix the bug of same timers are registered multiple times

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request will fix the bug of same timers are registered multiple times*

## Brief change log

- *deduplicate same timers in Python Operation*

## Verifying this change

- *Original Tests*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/pyflink/fn_execution/timerservice_impl.py [hunk 1]
```diff
@@ -59,6 +69,7 @@ def __init__(self, keyed_state_backend: RemoteKeyedStateBackend):
         self._keyed_state_backend = keyed_state_backend
         self._current_watermark = None
         self.timers = []  # type: List[Tuple[TimerOperandType, InternalTimer]]
+        self._exist_timers = set()  # type: Set[Tuple[TimerOperandType, InternalTimer]]
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/timerservice_impl.py:72]:
```suggestion
        self._existing_timers = set()  # type: Set[Tuple[TimerOperandType, InternalTimer]]
```

Comment (dianfu) [inline flink-python/pyflink/fn_execution/timerservice_impl.py:72]: What about change the type of timers to OrderedDict, then we can avoid to introduce _exist_timers.

File: flink-python/pyflink/fn_execution/timerservice_impl.py [hunk 2]
```diff
@@ -42,6 +42,16 @@ def get_key(self) -> K:
     def get_namespace(self) -> N:
         return self._namespace
 
+    def __hash__(self):
+        result = int(self._timestamp ^ (self._timestamp >> 32))
```
Comment (dianfu) [inline flink-python/pyflink/fn_execution/timerservice_impl.py:46]: What about cached the calculated hash value?

**Comment by HuangXingBo** [comment]
The failed test has been recorded in https://issues.apache.org/jira/browse/FLINK-20723 which is irrelevant to this PR.

**Review by dianfu — APPROVED** [review]

**Pull request closed without merging**